### PR TITLE
[Impeller] use DefaultUniformAlignment for compute buffers.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/144959

locally this only shows failures before on the impeller_goldens test shard.

### CODE REVIEW

File: impeller/entity/geometry/point_field_geometry.cc [hunk 1]
```diff
@@ -156,8 +156,8 @@ GeometryResult PointFieldGeometry::GetPositionBufferGPU(
       host_buffer.Emplace(points_.data(), points_.size() * sizeof(Point),
                           DefaultUniformAlignment());
 
-  BufferView geometry_buffer =
-      host_buffer.Emplace(nullptr, total * sizeof(Point), alignof(Point));
+  BufferView geometry_buffer = host_buffer.Emplace(
+      nullptr, total * sizeof(Point), DefaultUniformAlignment());
```
Comment (chinmaygarde) [inline impeller/entity/geometry/point_field_geometry.cc:160]: This should be the max of the alignof the Point and the default uniform alignment right?

**Comment by jonahwilliams** [comment]
Turning on validation to see which ones fail, then I'll add a deny list like goldens has.

File: impeller/entity/geometry/point_field_geometry.cc [hunk 2]
```diff
@@ -185,8 +185,8 @@ GeometryResult PointFieldGeometry::GetPositionBufferGPU(
   }
 
   if (texture_coverage.has_value() && effect_transform.has_value()) {
-    BufferView geometry_uv_buffer =
-        host_buffer.Emplace(nullptr, total * sizeof(Vector4), alignof(Vector4));
+    BufferView geometry_uv_buffer = host_buffer.Emplace(
+        nullptr, total * sizeof(Vector4), DefaultUniformAlignment());
```
Comment (chinmaygarde) [inline impeller/entity/geometry/point_field_geometry.cc:189]: Ditto about the max.

**Comment by jonahwilliams** [comment]
Done in https://github.com/flutter/engine/pull/51341

File: impeller/entity/geometry/point_field_geometry.cc (hunk 2 above)
Comment (jonahwilliams) [reply impeller/entity/geometry/point_field_geometry.cc:189]: Done in https://github.com/flutter/engine/pull/51341

File: impeller/entity/geometry/point_field_geometry.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/entity/geometry/point_field_geometry.cc:160]: Done in https://github.com/flutter/engine/pull/51341

**Pull request closed without merging**